File: javascript/867.Transpose-Matrix.js
/**
 * @param {number[][]} matrix
 * @return {number[][]}
 */
var transpose = function (matrix) {
  const rows = matrix.length;
  const cols = matrix[0].length;
  const result = [];

  for (let i = 0; i < cols; i++) {
    result[i] = Array(rows);
  }

  for (let i = 0; i < cols; i++) {
    for (let j = 0; j < rows; j++) {
      result[i][j] = matrix[j][i];
    }
  }

  return result;
};

File: python/867.Transpose-Matrix.py
class Solution:
    def transpose(self, matrix: List[List[int]]) -> List[List[int]]:
        rows = len(matrix)
        cols = len(matrix[0])

        result = [None] * cols

        for i in range(cols):
            result[i] = [None] * rows

        for i in range(cols):
            for j in range(rows):
                result[i][j] = matrix[j][i]

        return result